modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2675,15 +2675,17 @@ def broadcast_apply_full_axis(
             enumerate_partitions=enumerate_partitions,
             keep_partitioning=keep_partitioning,
         )
-        # Index objects for new object creation. This is shorter than if..else
-        kw = self.__make_init_labels_args(new_partitions, new_index, new_columns)
+        kw = {}
         if dtypes == "copy":
             kw["dtypes"] = self._dtypes
-        elif dtypes is not None:
+        elif dtypes is not None and new_columns is not None:
+            columns = new_columns if new_columns is not None else self.columns
```

Comment:
`self.columns` generally are not equal to new cols
```suggestion
            columns = new_columns if new_columns is not None else compute_new_columns()
```

Reply:
fixed